BE/app/rag_graph/node.py

```diff
@@ -0,0 +1,213 @@
+        model = gemini.model.with_structured_output(output_structure.RefineQuestion)
+
+        result = await model.ainvoke(input_prompt)
+        rewritten_question = result.rewritten_question
+        return GraphState(question=rewritten_question)
+    except Exception as e:
+        raise CustomException(
+            exception_case=ExceptionCase.GRAPH_NODE_ERROR,
+            detail=f"Error occured in node_refine_question: {e}",
+        )
+
+
+async def node_decision_context_need(state: GraphState) -> GraphState:
+    """
+    2. 쿼리를 보고 컨텍스트 검색이 필요한지 결정하는 노드.
+    """
+    try:
+        print("-------------------------")
+        print(node_decision_context_need.__name__)
+        print(state)
+        gemini = GeminiService(model="gemini-2.0-flash-lite")
+        question = state["question"]
+
+        input_prompt = prompt.check_context_need(question=question)
+        model = gemini.model.with_structured_output(output_structure.RouteQuery)
+
+        result = await model.ainvoke(input_prompt)
+        decision = result.decision
+        if decision == "context required":
+            return GraphState(is_context_need=True)
+        else:
+            return GraphState(is_context_need=False)
+    except Exception as e:
+        raise CustomException(
+            exception_case=ExceptionCase.GRAPH_NODE_ERROR,
+            detail=f"Error occured in node_decision_context_need: {e}",
+        )
+
+
+def router_decision_context_need(state: GraphState) -> bool:
+    """
+    검색이 필요 여부에 따라 다음 노드를 결정하는 라우터.
+    """
+    print("-------------------------")
+    print(router_decision_context_need.__name__)
+    print(state)
+
+    return state["is_context_need"]
+
+
+async def node_retriever(state: GraphState) -> GraphState:
+    """
+    3. 쿼리를 바탕으로 컨텍스트를 가져오는 노드.
+    """
+    try:
+        print("-------------------------")
+        print(node_retriever.__name__)
+        print(state)
+        question = state["question"]
+        authority_group = state["authority_group"]
+
+        embedding = await gemini.generate_embedding(
+            contents=question, task="RETRIEVAL_QUERY"
+        )
+        output_documents = await qdrant.query_document(
+            embedding=embedding,
+            metadata=DocumentMetadata(authority_group=authority_group),
+        )
+        documents = [
+            Document(
+                content=output_document.metadata.content,
+                datasource=output_document.metadata.datasource,
+                updated_at=output_document.metadata.updated_at,
+                page_id=output_document.metadata.page_id,
+            )
+            for output_document in output_documents
+        ]
+
+        return GraphState(context=documents)
+    except Exception as e:
+        raise CustomException(
+            exception_case=ExceptionCase.GRAPH_NODE_ERROR,
+            detail=f"Error occured in node_retriever: {e}",
+        )
+
+
+async def node_check_context_latest(state: GraphState) -> GraphState:
```

Comment:
함수명은 `{행위}_{대상}_{기타}` 형식이면 어떨까요?
```suggestion
async def check_latest_node_context(state: GraphState) -> GraphState:
```

Reply:
[ [922d1c4](https://github.com/f-lab-edu/AIagent-MLE/pull/5/commits/922d1c429fc4324838d2a50d18e81bc78124dded) ]

노드명과 엣지 코드 수정했습니다!